$NetBSD: patch-de,v 1.1 2011/05/21 15:33:00 wiz Exp $

pthread_key_create returns 0 on success; return values on failure are
not defined.

--- src/pulsecore/thread-posix.c.orig	2010-06-13 14:15:54.000000000 +0000
+++ src/pulsecore/thread-posix.c
@@ -172,7 +172,7 @@ pa_tls* pa_tls_new(pa_free_cb_t free_cb)
 
     t = pa_xnew(pa_tls, 1);
 
-    if (pthread_key_create(&t->key, free_cb) < 0) {
+    if (pthread_key_create(&t->key, free_cb)) {
         pa_xfree(t);
         return NULL;
     }
